collection/zset/zset_test.go

```diff
@@ -493,6 +493,30 @@ func TestFloat64SetRemoveRangeByRank(t *testing.T) {
 	assert.Equal(t, N, z.Len()+len(actualNs))
 }
 
+func TestFloat64SetRemoveRangeByRankConcurrently(t *testing.T) {
+	const N = 10000
+	z := NewFloat64()
+	for i := 0; i < N; i++ {
+		z.Add(fastrand.Float64(), fmt.Sprint(i))
+	}
+
+	start, stop := func(a, b int) (int, int) {
+		if a < b {
+			return a, b
+		} else {
+			return b, a
+		}
+	}(fastrand.Intn(N), fastrand.Intn(N))
+
+	const G = 10
+	for i := 0; i < G; i++ {
+		go func() {
+			z.RemoveRangeByRank(start, stop)
+		}()
+	}
+	time.Sleep(3 * time.Second)
```

Comment:
We can use sync.WaitGroup to ensure that all goroutines exit properly.

Reply:
fixed